src/pool/cli.c

```diff
@@ -2783,3 +2783,26 @@ dc_pool_stop_svc(tse_task_t *task)
 	tse_task_complete(task, rc);
 	return rc;
 }
+
+int dc_pool_get_redunc(daos_handle_t poh)
+{
+	struct daos_prop_entry	*entry;
+	daos_prop_t		*prop_query = daos_prop_alloc(1);
+	int			rf;
+
+	if (prop_query == NULL)
+		return -DER_NOMEM;
+
+	prop_query->dpp_entries[0].dpe_type = DAOS_PROP_PO_REDUN_FAC;
+	rf = daos_pool_query(poh, NULL, NULL, prop_query, NULL);
```

Comment:
Originally i hit problem calling sync in async, so i moved this out from task, see daos_cont_inherit_redunc_fac() is called in daos_cont_create2() that happens before dc_task_create(), so it probably ok?.

Extend pool_connect_out to return more pool information could break compatibity between older clients and new server, i don't have strong objection for that though.

Reply:
I think that generally make sync call with RPC within async API is not good, and it could affect cont_create's performance.
But probably fine to leave it to be resolved in following other patch, I don't want to block this patch's land.

Reply:
make sense, thanks!

Reply:
np, to avoid RPC protocol change, probably can do another pool query inside client's pool connect, so can cache that prop and need not query again later. just FYI.

Reply:
I pushed a simple commit to cache rf prop, but i found this caching only work for api which pool handle can be shared inside same process etc, i am wondering to make pool connect more extendable, it might make sense to introduce a POOL_CONNECT_V2.